### ISSUE TITLE
look into using arping instead of ping

### ISSUE THREAD

**Issue by jorijnsmit** [jorijnsmit/onlineChecker#2]
Is it more reliable? Could a combination of both commands be the most stable solution?

**Comment by betyar** [comment]
I find arping more reliable. On my Samsung Galaxy ping doesn't work but arping is no problem.

**Comment by robzgit** [comment]
Android 9 and iphone iOS 12 and 13 does not react to ping most of the time. Almost only when screen is unlocked! Not usable.
 
sudo arping -c 10 -I eth0 192.168.1.1 does work much better. I get sometimes 10 out of 10 pongs but it can also be like this (iphone with black screen) real world example:

pi@smartmtrpi:~/domoticz/scripts/python$ sudo arping -c 100 -I eth0 bla
ARPING 192.168.253.204
60 bytes from bla: index=0 time=26.538 msec
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
60 bytes from bla: index=1 time=83.689 msec
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
60 bytes from bla: index=2 time=42.941 msec
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
Timeout
60 bytes from bla: index=3 time=102.186 msec
Timeout
Timeout
Timeout
^C
--- bla statistics ---
39 packets transmitted, 4 packets received,  90% unanswered (0 extra)

"bla is IP address or mac and IP address".

My conclusion after some hours of arping that you need sets of 10 arpings or always running script to get enough answers.

**Comment by Wellknown4** [comment]
yes it is not working also my phone with black screen seems to cut the wifi, so there is no one home, it thinks, everything goes off, the heating, the lights, i sit in the dark. So i have to keep my screen on, then it works.,

haha

**Comment by zeddski** [comment]
> 
> 
> Is it more reliable? Could a combination of both commands be the most stable solution?

I find it more reliable. I'm the one who altered a previous version of your script to support arping and have been using that ever since.
To be honest, I've only to content with Android devices here. With two exceptions (my mother's phone and my mother-in-law's), it works flawlessly.

You might offer a setting for people to decide which they want to use, depending on what works best for their environment?
In my version I use 
`'sudo arping -q -c4 -W 1 '+ device + ' > /dev/null'`

**Comment by PatrickStel** [comment]
I'm using this script to check if my Smart TV is on or not, so I had to change arping back to ping.

Cause arping uses the link layer and my network interface in my tv is always listening and replying even if the Tizen OS is off.

So i modified it a bit 
`samsungtv=sys.argv[5]
 if samsungtv == 'yes':
    currentstate = subprocess.call('sudo ping -q -c1 -W 1 '+ device + ' > /dev/null', shell=True)
  else:
    currentstate = subprocess.call('sudo arping -q -c1 -W 1 '+ device + ' > /dev/null', shell=True)`
